slice/internal/controller/workload_controller.go

```diff
@@ -100,22 +107,14 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 	log.V(3).Info("Reconcile Workload")
 
 	if finalize, reason := shouldFinalize(wl); finalize {
-		if controllerutil.ContainsFinalizer(wl, SliceControllerName) {
-			log.V(3).Info(fmt.Sprintf("Cleaning up the Slice and finalize the Workload because %s", reason))
-			err = r.client.Delete(ctx, core.SliceWithMetadata(wl))
-			if client.IgnoreNotFound(err) != nil {
-				return ctrl.Result{}, err
-			}
-			controllerutil.RemoveFinalizer(wl, SliceControllerName)
-			if err := r.client.Update(ctx, wl); err != nil {
-				if !apierrors.IsNotFound(err) {
-					log.Error(err, "Failed to remove finalizer")
-				}
-				return ctrl.Result{}, client.IgnoreNotFound(err)
-			}
-			log.V(3).Info("Removed finalizer")
+		log.V(3).Info(fmt.Sprintf("Cleaning up the Slice and finalize the Workload because %s", reason))
+		cleanedUp, err := r.cleanupSlice(ctx, wl)
+		if err != nil || !cleanedUp {
+			return ctrl.Result{}, err
 		}
```

Comment:
perhaps RequeueAfter here, for the !cleanedUp branch?

Reply:
Done